subprojects/core/src/main/java/org/gradle/cache/internal/DefaultCrossBuildInMemoryCacheFactory.java

```diff
@@ -58,8 +59,8 @@ public DefaultCrossBuildInMemoryCacheFactory(ListenerManager listenerManager) {
     }
 
     @Override
-    public <K, V> CrossBuildInMemoryCache<K, V> newCacheRetainingDataFromPreviousBuild() {
-        CrossBuildCacheRetainingDataFromPreviousBuild<K, V> cache = new CrossBuildCacheRetainingDataFromPreviousBuild<>();
+    public <K, V> CrossBuildInMemoryCache<K, V> newCacheRetainingDataFromPreviousBuild(Predicate<V> keepBetweenBuilds) {
+        CrossBuildCacheRetainingDataFromPreviousBuild<K, V> cache = new CrossBuildCacheRetainingDataFromPreviousBuild<>(keepBetweenBuilds);
```

Comment:
I'd prefer a noun-like name, like `retentionFilter` here. WDYT?

Reply:
Renamed to `retentionFilter`.